torch/csrc/lazy/core/shape.cpp

```diff
@@ -33,5 +33,14 @@ std::vector<Shape> convertShapes(
   return shape;
 }
 
+size_t Shape::numel() const {
+  size_t elts = dim() > 0 ? 1 : 0;
+  for (size_t i = 0; i < dim(); i++) {
```

Comment:
A range-based loop like `for (auto size : sizes_)` would be cleaner, since this is a method of Shape (meaning can access sizes_).

One reminder is to remember to merge changes under torch/csrc/lazy into master at some point.